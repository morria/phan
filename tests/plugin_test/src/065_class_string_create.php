<?php
/**
 * @template T
 * @param class-string<T> $name
 * @return T
 */
function create_class(string $name) {
    return new $name;
}
// Here, we con
echo strlen(create_class(0));  // should warn
echo strlen(create_class('stdClass'));  // should infer that this creates an instance of stdClass (and warn about it being invalid for strlen)
echo strlen(create_class('invalidclass'));  // should warn
echo strlen(create_class('int'));  // should warn
echo strlen(create_class(''));  // should warn
