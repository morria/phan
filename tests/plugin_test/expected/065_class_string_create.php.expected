src/065_class_string_create.php:12 PhanTypeMismatchArgumentInternal Argument 1 (string) is \stdClass but \strlen() takes string
src/065_class_string_create.php:13 PhanTypeMismatchArgumentInternal Argument 1 (string) is \invalidclass but \strlen() takes string
src/065_class_string_create.php:13 PhanUndeclaredClassReference Reference to undeclared class \invalidclass
src/065_class_string_create.php:14 PhanTypeMismatchArgumentInternal Argument 1 (string) is \int but \strlen() takes string
src/065_class_string_create.php:14 PhanUndeclaredClassReference Reference to undeclared class \int
src/065_class_string_create.php:15 PhanEmptyFQSENInClasslike Possible use of a classlike '' with an empty FQSEN.
